hathor/utils/arg_parse.py

```diff
@@ -0,0 +1,24 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from argparse import Namespace
+from typing import NamedTuple, Type, TypeVar
+
+from hathor.utils.named_tuple import validated_named_tuple_from_dict
+
+T = TypeVar('T', bound=NamedTuple)
+
+
+def from_namespace(named_tuple_type: Type[T], namespace: Namespace) -> T:
+    return validated_named_tuple_from_dict(named_tuple_type, vars(namespace))
```

Comment:
Is this in preparation for a generic use in the future? Otherwise wouldn't it be simpler to put this directly in `RunNodeArgs`? Like:

```python
class RunNodeArgs(NamedTuple):
    # ...
    @classmethod
    def from_namespace(cls, namespace: Namespace) -> 'RunNodeArgs':
        return validated_named_tuple_from_dict(cls, vars(namespace))
```

Reply:
Yes, I put it there thinking about migrating all CLI commands to this new approach. Do you think it's overkill?

Reply:
It makes sense, but also, they could inherit from `RunNodeArgs` and their custom parameters there. So the `from_namespace` would be re-used. Or for using it even on non-"run-node" subcommands, we could even do a ` class BaseArgs(NamedTuple)` class with just the `from_namespace` classmethod. But I'm OK leaving it as is for now and changing it when we get to implement it in the future.